### PR TITLE
Removed PHP shorttags and add new ids endpoint

### PR DESCRIPTION
**Pull request by bergrans**
By adding some missing PHP tags the Sensor(s) drop-down work like it should (on my local Lando environment).

### CODE REVIEW

File: data/ids/index.php
Comment (matthijskooijman) [inline data/ids/index.php]: Why put these ranges in a separate list and parse them later? Why not just parse them right away here? That would be a bit simpler, and preserves the order of ids as listed in the json file as well?

Comment (matthijskooijman) [reply data/ids/index.php]: Oh, I see you sort the list later, but still, keeping the order might make sorting faster :-)

File: data/ids/index.php
Comment (matthijskooijman) [inline data/ids/index.php]: I wonder if `getProjects` is really needed? Why not just do a check with e.g. `array_key_exists` here in the function, rather than going through the motions of creating an `array_keys` array, and then doing `in_array` on it? The code you wrote works, but I think it can be signficantly shorter and simpler?

**Review by matthijskooijman — CHANGES_REQUESTED** [review]
Thanks for this contribution, sounds like a good addition to the API.

Overall, the code looks good and well-documented. I left a few remarks inline where I think the code can be a lot simpler.

One additional remark: you now do lowercase matching of the project name, but AFAICS this relies on the project names being lowercase in sensorsets.json. This is currently the case, but to prevent issues when ids are added with uppercase letters (with the current code, I cannot select a project called "Foo" at all), I would suggest to either:
 - Make both the GET argument and the project name from JSON lowercase before comparison, for true case insensitive matching.
 - Just do case sensitive matching and let people just use the right project id.

I would be inclined to the latter approach, since it has less wiggleroom that might byte you later :-)

**Comment by bergvdm** [comment]
Implemented some of the changes you suggested to optimize the code.

**Comment by matthijskooijman** [comment]
Thanks. Seems I missed your comment about this, so only looked at the code now. I think the code itself is perfect now, thanks for picking up my suggestions (and adding some more improvements). Before I merge, I'd like to:
 - See the improvement commits squashed (leaving only two commits, I think, one to fix the PHP shorttags and one adding the new script).
 - Have this rebased on top of master.

If you want me to do this (not sure how familar with git you are), let me know, it's little effort.

**Comment by bergrans** [comment]
It might be a good idea if you squash and rebase the commits the way you want them to be. Next time I will use proper feature branches, sorry for the inconvenience.

**Comment by matthijskooijman** [comment]
> It might be a good idea if you squash and rebase the commits the way you want them to be. Next time I will use proper feature branches, sorry for the inconvenience.

No inconvenience at all! I'm note sure what "proper feature branches" entails exactly (maybe one branch per feature so it can be squash-merged?), but I personally rather like the approach where you just add some additional commits during development of a PR, showing a little bit of extra history (I often use `git commit --fixup` for this), and then squashing/rebasing them into proper commits shortly before merging (or also more often halfway once changes have been reviewed). Doing a squash-merge in Github is fine for very small single-change PRs, but putting multiple changes in the same PR is IMHO also fine, if they're (eventually) properly separated in different commits (even highly related changes might be better put in different commits sometimes).


Anyway, I'll squash and merge right away :-)

**Comment by matthijskooijman** [comment]
Ok, merged. Note that I did a force push to your master branch (where you created this PR from), so you might want to reset your local branch to there as well or so.

**Pull request merged**
